sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -178,7 +168,7 @@ def close(reason):
                     owner_id, eventhub_name, partition_id, consumer_group_name, reason
                 )
                 try:
-                    self._partition_close_handler(partition_context, reason)
+                    self._callback_queue.put((self._partition_close_handler, partition_context, reason), block=True)
```

Comment:
If we set `block=True` then wont the client hang if we get more than 10000 events?

Reply:
This comment applies to all the `queue.put()` operations

Reply:
Ah I see - this code is running in a background thread